src/browser/components/molecules/TaskList/index.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react'
+import Layout from '../../molecules/Layout/index'
+import Task from '../Task/index'
+import styles from './index.css';
+const moment = require('moment')
+
+export default class TaskList extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      tasks: []
+    }
+  }
+
+  componentWillMount(){
+    this.loadTasks()
+  }
+
+  loadTasks(){
+    const fetchDetails = {
+      method: 'GET',
+      mode: 'cors',
+      headers: new Headers({
+        'Accept': 'application/json, application/xml, text/plain, text/html, *.*',
+        'Content-Type': 'application/json',
+         }),
+      credentials: 'same-origin'
+    }
+    fetch('http://noob.learnersguild.dev/api/task', fetchDetails)
+    .then( data => {
+      return data.json()
+    })
+    .then( result => {
+      this.setState({
+        tasks: result.sort((a,b) => {
+          if( moment(a.due_date).isBefore(b.due_date) ){
+            return -1
+          } else {
+            return 1
+          }
+        })
+      })
+    })
+    .catch( err => {
+      console.log('Error loading tasks', err);
+      return err
+    })
+  }
+
+  render() {
```

Comment:
the `render` function is doing too much. I would like functions which do too much to be broken down to smaller functions, which typically do only 1 thing. Also I'm a fan of the functional paradigm, so I would advice against using `for` loops. I would use the `groupBy` function(available is `lodash/groupBy` to group tasks by `is_complete`. 

Something like this -
```
  import groupBy from 'lodash/groupBy'

  tasksJSX(tasks) {
    return tasks.map(task => 
                   <Task
                     key={task.id}
                     id={task.id}
                     body={task.body}
                     loadTasks={this.loadTasks.bind(this)}
                     due_date={task.due_date}
                     completed_on={task.updated_at}
                     is_complete={task.is_complete}/>)
  }

  render() {
    const tasksGroupedByComplete = groupBy(this.state.tasks,
                                           (task) => task.is_complete)
    const inProgressTasks = tasksGroupedByComplete.false || []
    const completedTasks = tasksGroupedByComplete.true || []
    return (
        <div className={styles.TaskList}>
          <div>Tasks In Progress</div>
          <div> {this.tasksJSX(inProgressTasks)} </div>
          <br/>
          <div>
            <div>Completed Tasks</div>
            {this.tasksJSX(completedTasks)}
          </div>
        </div>
    )
  }
```